### DISCUSSION TITLE
auto-inject of styles is not done when scripts are manually added

### DISCUSSION THREAD

**Discussion by woodspire** [livewire/livewire#7414] [7. Bugs (archived)]
### Livewire version

v3 latest

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

manually inject scripts via @livewireScripts

the styles is not auto-injected.

adding @livewireStyles works, but it defeat the purpose of auto-injeection

### Additional information

https://github.com/livewire/livewire/blob/6d39f1f583188b168abaf58947a9202be97818d0/src/Features/SupportAutoInjectedAssets/SupportAutoInjectedAssets.php#L61

There is only a checkup for scripts, not styles.

Both should be treated independently. 

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by woodspire** [comment]
I discovered this bug this way:

auto inject to true

but I wanted alpine on a non-livewire webpage

add livewivescripts globally

alpine worked.


later find out that wire:loading was not hidden (done by styles)

**Comment by woodspire** [comment]
solution:

explain in doc that manually added one of the other disable the whole auto injecting.

or fix code and add test

the method that check can still have the same name and return true when AT LEAST ONE item needs to be injected.

Then in the injection procedure, detect what needs to be injected. If nothing is to be injected throw an exception because it should not have been called (it means both methods do not evaluate the same stuff)

**Comment by woodspire** [comment]
![image](https://github.com/livewire/livewire/assets/1275609/817cff39-ce5d-4175-a388-9cde8688a359)

**Reply by woodspire** [reply]
It does not explain that the styles that WERE autoinjected won't be.

**Comment by woodspire** [comment]
@calebporzio I could do a PR to fix this. Any problem about that ?

**Reply by joshhanley** [reply]
@woodspire All PRs are welcome 🙂
